eng/common/LiveTestResources/New-TestResources.ps1.md

```diff
@@ -0,0 +1,392 @@
+---
```

Comment:
Given I already authored script help directly in the scripts (which `get-help` supports), what is the purpose of the markdown files? Do they get published to a web site for online consumption? Where? How do customers find them?

Reply:
This PR could use some more context.. Happy to chat more. 

This is the first step in a process to update these scripts across our repositories and provide requested documentation. 

1. Synchronize these scripts across repos (https://github.com/Azure/azure-sdk-tools/issues/358) -- this PR puts this file into eng\common which will synchronize it to other repos. 
1. Provide documentation describing how to run (https://github.com/Azure/azure-sdk/issues/1021) -- This PR does that
1. Merge synchronized changes into respective repos (out of scope for this PR)
1. Update CONTRIBUTING.md files to link to the documentation. These links can stay relative to the repo because the previous step merged the markdown files into each repo. (out of scope for this PR) 
1. Update live test yaml files to use these scripts. (out of scope for this PR)

This PR includes some changes to the original `Get-Help` documentation comments including an [example](https://github.com/Azure/azure-sdk-tools/pull/410/files#diff-540557022506777cbff836b814e495d1R52) which details what steps a contributor could take to deploy resources when developing on their desktop. Most of what was originally in the documentation is still there... the intent here was not to rewrite the useful parts of what had been written.

**What is the purpose of the markdown files?**

The purpose of the Markdown files is to provide this documentation in a legible format which is linkable from CONTRIBUTING.md. Generating markdown files from the `Get-Help` comments is cheap and keeps the `Get-Help` comments as the source of truth for how to use these scripts. 

**Do they get published to a web site for online consumption? Where?**

Changes to this folder in this repo are replicated to folders in subscribed repos (e.g. .NET, Java, JS, Python) so changes to these documentation files will be synchronized into subscribed repos. 

**How do customers find them?**

CONTRIBUTING.md files in each of the language repos where these scripts are used can be updated to link to these files. Changes to this documentation should happen once rather than trying to synchronize changes across all of the repos. Changes to CONTRIBUTING.md files will have to happen in separate PRs.